### PR TITLE
Reviews#8

### PR DESCRIPTION
**Pull request by Jacoby5770**
Working on #8. Got most of the routes and components done with reviews. Have to hook it up to individual products now.

### CODE REVIEW

File: client/components/AddReview.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    postReview: (newReview) => dispatch(postReview(newReview))
+  }
+}
+
+export default connect(null, mapDispatchToProps)(AddReview)
+
```
Comment (pckilgore) [inline client/components/AddReview.js]: :+1:

File: client/components/EditProduct.js [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
 }
 
-export default EditProducts
+const mapDispatchToProps = dispatch => ({
+  updateProductById: product => dispatch(updateProductById(product))
+})
+
+export default connect(null, mapDispatchToProps)(EditProducts)
+
+
```
Comment (pckilgore) [inline client/components/EditProduct.js]: Not sure I understand why this component was edited?

These look like old changes.  Try to rebase (pull in commits from master) before you submit a PR next time.
```
$ git pull origin master
(figure out how to merge)
$ git commit (merge changes)
$ git push
```

File: client/components/EditReview.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }
+}
+
+const mapDispatchToProps = dispatch => ({
+  updateReviewById: review => dispatch(updateReviewById(review))
+})
+
+export default connect(null, mapDispatchToProps)(EditReview)
+
+
```
Comment (pckilgore) [inline client/components/EditReview.js]: This code looks fine, but editing doesn't appear to be working when I try to test it?  It looks like the route isn't wired up?

File: client/components/ReviewForm.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          // M.AutoInit()
+          M.updateTextFields()
+          M.textareaAutoResize(document.getElementById('description'))
+        }, 1)}
+      </script>
+    </div>
+  )
+}
+
+const mapDispatchToProps = dispatch => ({})
```
Comment (pckilgore) [inline client/components/ReviewForm.js]: Looks like this line isn't necessary

File: client/components/ReviewForm.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+const mapStateToProps = (state, {match}) => ({
+  // This `initialValues` variable name below is required by redux-forms
+  initialValues: state.reviews.byId[match.params.reviewId]
+})
+
+ReviewForm = reduxForm({form: 'reviewFrom'})(ReviewForm)
+ReviewForm = connect(mapStateToProps, mapDispatchToProps)(ReviewForm)
```
Comment (pckilgore) [inline client/components/ReviewForm.js]: This can be change from:
```javascript
ReviewForm = connect(mapStateToProps, mapDispatchToProps)(ReviewForm)
```
to:
```javascript
ReviewForm = connect(mapStateToProps)(ReviewForm)
```
(assuming I'm correct and we don't need a mapToDispatch fn above.

File: client/routes.js [hunk 6]
```diff
@@ -40,6 +43,8 @@ class Routes extends Component {
           <Switch>
             <Route exact path="/product/add" component={AddProduct} />
             <Route path="/product/:productId/edit" component={EditProduct} />
+            <Route path="/review/add" component={(AddReview)} />
+            {/* <Route path="/review/:reviewId/edit" component={EditReview} /> */}
```
Comment (pckilgore) [inline client/routes.js]: The edit route is disabled here.  Let's fix editing before we merge into master.

File: client/store/reviews.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        byId: {
+          ...state.byId,
+          [action.updatedReview.id]: action.updatedReview
+        },
+        allIds: [...state.allIds]
+      }
+    default:
+      return state
+  }
+}
```
Comment (pckilgore) [inline client/store/reviews.js]: This all looks good.

File: server/api/reviews.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+const router = require('express').Router()
+const {Review} = require('../db/models')
+module.exports = router
+
+const createReviewFromJSON = ele => ({
+  rating: '' + ele.rating,
```
Comment (pckilgore) [inline server/api/reviews.js]: Isn't rating an integer?

File: server/db/models/product.js [hunk 9]
```diff
@@ -1,4 +1,3 @@
-const Category = require('./category')
```
Comment (pckilgore) [inline server/db/models/product.js]: Why is this deleted??

File: script/ReviewsData.json [hunk 10]
```diff
@@ -0,0 +1,32 @@
+[
+    {
+      "rating": 1,
+      "body": "Nulla ut erat id mauris vulputate elementum. Nullam varius. Nulla facilisi.\n\nCras non velit nec nisi vulputate nonummy. Maecenas tincidunt lacus at velit. Vivamus vel nulla eget eros elementum pellentesque.\n\nQuisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus."
```
Comment (pckilgore) [inline script/ReviewsData.json]: I think we need to also seed what products reviews belong to, and what users reviews belong to.

**Comment by Jacoby5770** [comment]
...be able to leave reviews for products (including text and a 5-star rating), so that I can share my experiences with other visitors

this is the requirement

File: server/api/reviews.js (hunk 8 above)
Comment (Jacoby5770) [reply server/api/reviews.js]: yes

File: server/db/models/product.js (hunk 9 above)
Comment (Jacoby5770) [reply server/db/models/product.js]: Isn't used in model. Changed it back.

File: script/ReviewsData.json (hunk 10 above)
Comment (Jacoby5770) [reply script/ReviewsData.json]: added

File: client/components/ReviewForm.js (hunk 4 above)
Comment (Jacoby5770) [reply client/components/ReviewForm.js]: I don't think it is, but you mentioned something when working on ProductForm not working when it wasn't there

Comment (Jacoby5770) [reply client/components/ReviewForm.js]: I deleted it for now

File: client/components/EditReview.js (hunk 3 above)
Comment (Jacoby5770) [reply client/components/EditReview.js]: The route was commented out. I uncommented it.

**Comment by Jacoby5770** [comment]
Got reviews to work (add/edit reviews). Right now, once you add a review, the page routes to the all reviews page instead of the single product page. Also, I cannot figure out anchor tags. I included the id for each review and included the review id in the history push, but the page still just routes to the top of the reviews.
